app/code/community/Bolt/Boltpay/controllers/OnepageController.php

```diff
@@ -14,79 +14,47 @@
-                        } catch (Mage_Core_Exception $e) {
-                            $result['error'] = $e->getMessage();
-                            Mage::helper('boltpay/bugsnag')->notifyException($e);
-                        } catch (Exception $e) {
-                            //Mage::logException($e);
-                            $result['error'] = $this->__('Unable to set Payment Method.');
-                            Mage::helper('boltpay/bugsnag')->notifyException($e);
-                        }
-                    }
+        $lastQuoteId = $session->getLastQuoteId();
+        $lastOrderId = $session->getLastOrderId();
+        $lastRecurringProfiles = $session->getLastRecurringProfileIds();
+        if (!$lastQuoteId || (!$lastOrderId && empty($lastRecurringProfiles))) {
+            $this->_redirect('checkout/cart');
+            return;
+        }
 
-                    if (!$result) {
-                        Mage::dispatchEvent(
-                            'checkout_controller_onepage_save_shipping_method',
-                            array(
-                                'request' => $this->getRequest(),
-                            'quote' => $this->getOnepage()->getQuote())
-                        );
-                        Mage::helper('boltpay')->collectTotals($this->getOnepage()->getQuote());
-                        $this->getResponse()->setBody(Mage::helper('core')->jsonEncode($result));
-                        if (Mage::getStoreConfig('payment/boltpay/skip_payment')) {
-                            $this->loadLayout('checkout_onepage_review');
-                            $result['goto_section'] = 'review';
-                            $result['update_section'] = array(
-                                'name' => 'review',
-                                'html' => $this->_getReviewHtml());
-                        } else {
-                            $result['goto_section'] = 'payment';
-                            $result['update_section'] = array(
-                                'name' => 'payment-method',
-                                'html' => $this->_getPaymentMethodsHtml());
-                        }
-                    }
-                }
+        $session->clear();
+        $this->loadLayout();
+        $this->_initLayoutMessages('checkout/session');
 
-                Mage::helper('boltpay')->collectTotals($this->getOnepage()->getQuote())->save();
-                $this->getResponse()->setBody(Mage::helper('core')->jsonEncode($result));
-            }
-        } catch (Exception $e) {
+        try {
+            Mage::dispatchEvent('checkout_onepage_controller_success_action', array('order_ids' => array($lastOrderId)));
+        } catch ( Exception $e ) {
             Mage::helper('boltpay/bugsnag')->notifyException($e);
-            throw $e;
+            
+            if (strtolower($this->_getOrder()->getPayment()->getMethod()) !== Bolt_Boltpay_Model_Payment::METHOD_CODE) {
```

Comment:
hmm is it safe to swallow error? can you give me an example of case we want to ignore exception?

Reply:
I think we would better let them go. At this point the bolt plugin actually already complete its task and order is alread successfully placed, and even there is any exception from other plugins, we would better let them handle in their own way.

Reply:
@daisy1754 
>At this point the bolt plugin actually already complete its task 
>and the order is already successfully placed

The only thing left for Bolt to do on Bolt orders is to show the order success screen to the end user.  Any error that comes up at this point throws an error screen to the end user, giving Bolt the appearance of being buggy and not succeeding.  

A real case for this is Indique Dev and possibly Indique live where certain plugins are performing post order save task that seeks conditions for their payment methods regardless of other payment methods being used.

In our code, for Bolt, we let our plugin succeed. For other payment methods or
> other plugins, we ... let them handle in their own way